### PR TITLE
Add support for PostgreSQL Insert table aliases (#1069)

### PR DESCRIPTION
**Pull request by boydjohnson**
This is a fix for #1069

### CODE REVIEW

File: src/ast/mod.rs [hunk 1]
```diff
@@ -2411,6 +2414,16 @@ impl fmt::Display for Statement {
                 replace_into,
                 priority,
             } => {
+                let table_name = if let Some(Ident {
+                    value,
+                    quote_style: _,
+                }) = table_alias
+                {
+                    format!("{table_name} AS {value}")
```
Comment (alamb) [inline src/ast/mod.rs:2422]: It seems like we should format the alias using whatever quote style was originally used

Maybe something like

```suggestion
                let table_name = if let Some(table_alias) = table_alias
                {
                    format!("{table_name} AS {table_alias}")
```

File: tests/sqlparser_postgres.rs [hunk 2]
```diff
@@ -3500,3 +3500,69 @@ fn parse_join_constraint_unnest_alias() {
...
     );
 }
+
+#[test]
+fn test_complex_postgres_insert_with_alias() {
+    let sql1 = "WITH existing AS (SELECT test_table.id FROM test_tables AS test_table WHERE (a = 12) AND (b = 34)), inserted AS (INSERT INTO test_tables AS test_table (id, a, b, c) VALUES (DEFAULT, 56, 78, 90) ON CONFLICT(a, b) DO UPDATE SET c = EXCLUDED.c WHERE (test_table.c <> EXCLUDED.c)) SELECT c FROM existing";
+
+    pg().verified_stmt(sql1);
+
+    let sql2 = "INSERT INTO test_tables AS test_table (id, a) VALUES (DEFAULT, 123)";
```
Comment (alamb) [inline tests/sqlparser_postgres.rs:3510]: Could you please:
1. Make the `sql2` a second test (e.g. `test_simple_postgres_insert_with_alias`)
2. Add a test for a quoted identifier, something like  `INSERT INTO test_tables AS "Test_Table" (id, a) VALUES (DEFAULT, 123)`, perhaps

**Review by alamb — COMMENTED** [review]
Thank you for this contribution @boydjohnson  - this is looking close.

File: src/ast/mod.rs (hunk 1 above)
Comment (boydjohnson) [reply src/ast/mod.rs]: Thanks, I followed your advice. I rebased and force-pushed. I hope that is ok.

**Review by alamb — APPROVED** [review]
Thank you @boydjohnson

**Pull request merged**